### PR TITLE
Rename ValkeyModule_DefragModuleString to ValkeyModule_DefragValkeyModuleString

### LINKED ISSUE #242: Rename ValkeyModule_DefragModuleString to ValkeyModule_DefragValkeyModuleString
In https://github.com/valkey-io/valkey/pull/194, we removed the Redis modifier from RedisModule_DefragRedisModuleString instead of updating it to Valkey. Since this is describing a ValkeyModuleString, it seems more important to keep valkey in the name. See https://github.com/search?q=repo%3Avalkey-io%2Fvalkey%20ValkeyModule_DefragModuleString&type=code for places we need to update.

### PR DESCRIPTION
**Pull request by 9bany**
fixes: https://github.com/valkey-io/valkey/issues/242

### CODE REVIEW

**Review by zuiderkwast — COMMENTED** [review]
Thanks @0del!

The function `VM_DefragModuleString` in `module.c` needs to be renamed too. It's the actual implementation of this.

Can you also fix this comment? https://github.com/valkey-io/valkey/blob/aaec321213605e6315d85f2e617b4f33420b8df7/src/valkeymodule.h#L7

It says "Symbolic names have been changed from VALKEYMODULE_* to VALKEYMODULE_* to align with the new module naming convention" but it should of course be "... changed from REDISMODULE_* to VALKEYMODULE_* ..." instead.

**Comment by 9bany** [comment]
Well done @zuiderkwast 
I fixed issue linking in this PR, as well as the issue in your comment

**Review by PingXie — APPROVED** [review]
Thanks!

File: src/valkeymodule.h [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * This header file is forked from redismodule.h to reflect the new naming conventions adopted in Valkey.
  * 
  * Key Changes:
- * - Symbolic names have been changed from VALKEYMODULE_* to VALKEYMODULE_* to align with the
+ * - Symbolic names have been changed from REDISMODULE_* to VALKEYMODULE_* to align with the
```
Comment (zuiderkwast) [inline src/valkeymodule.h]: Now I have a very easy chance to suggest changes, so I will suggest to something I think is even better. 😎

```suggestion
 * - Symbolic names have been changed from containing RedisModule, REDISMODULE, etc. to ValkeyModule, VALKEYMODULE, etc. to align with the
```

**Review by zuiderkwast — APPROVED** [review]

**Comment by madolson** [comment]
@0del Do you have a name you want to go by in the release notes? (Ideally your real name, but we're asking people to identify with more than just an alias)

**Comment by 9bany** [comment]
@madolson Sure, My name is Bany

**Pull request merged**